lib/errors.ts

```diff
@@ -126,56 +126,81 @@ const messages: {
 	`,
 
 	BalenaExpiredToken: () => stripIndent`
-		Looks like your session token is expired.
+		Looks like your session token has expired.
 		Please try logging in again with:
 			$ balena login`,
 };
 
+const EXPECTED_ERROR_REGEXES = [
+	/^BalenaApplicationNotFound:/, // balena-sdk
+	/^BalenaDeviceNotFound:/, // balena-sdk
+	/^Missing \w+$/, // Capitano, oclif parser: RequiredArgsError, RequiredFlagError
+	/^Unexpected arguments?:/, // oclif parser: UnexpectedArgsError
+	/to be one of/, // oclif parser: FlagInvalidOptionError, ArgInvalidOptionError
+];
+
 export async function handleError(error: any) {
+	// Set appropriate exitCode
 	process.exitCode =
 		error.exitCode === 0
 			? 0
 			: parseInt(error.exitCode, 10) || process.exitCode || 1;
 
+	// Handle non-Error objects (probably strings)
 	if (!(error instanceof Error)) {
 		printErrorMessage(String(error));
 		return;
 	}
 
+	// Prepare message
 	const message = [interpret(error)];
 
-	if (process.env.DEBUG && error.stack) {
-		message.push('\n' + error.stack);
-	}
-	printErrorMessage(message.join('\n'));
-
-	const expectedErrorREs = [
-		/^BalenaApplicationNotFound:/, // balena-sdk
-		/^BalenaDeviceNotFound:/, // balena-sdk
-		/^Missing \w+$/, // Capitano's command line parsing error
-		/^Unexpected arguments?:/, // oclif's command line parsing error
-	];
-
-	if (
-		error instanceof ExpectedError ||
-		expectedErrorREs.some(re => re.test(message[0]))
-	) {
-		return;
+	if (error.stack) {
+		if (process.env.DEBUG) {
+			message.push('\n' + error.stack);
+		} else {
+			// Include first line of stacktrace
+			message.push('\n' + error.stack.split(`\n`)[0]);
```

Comment:
Is this useful at all? I would assume we'd like to not show this completely without the DEBUG flag (hence my suggestion above)

Reply:
This was a feature request from @Page- during the discussion about changes.